docs/building-a-node-with-ldk/setting-up-a-channel-manager.md

```diff
@@ -0,0 +1,1276 @@
+let netGraph = NetworkGraph(network: network, logger: logger)
+
+// Initialise RGS
+let rgs = RapidGossipSync(networkGraph: netGraph, logger: logger)
+if let lastSync = netGraph.getLastRapidGossipSyncTimestamp(), let snapshot = getSnapshot(lastSyncTimeStamp: lastSync) {
+  let timestampSeconds = UInt64(NSDate().timeIntervalSince1970)
+  let res = rgs.updateNetworkGraphNoStd(updateData: snapshot, currentTimeUnix: timestampSeconds)
+  if res.isOk() {
+    print("RGS updated")
+  }
+} else if let snapshot = getSnapshot(lastSyncTimeStamp: 0) { // Use lastSyncTimeStamp as 0 for first Sync
+  let timestampSeconds = UInt64(NSDate().timeIntervalSince1970)
+  let res = rgs.updateNetworkGraphNoStd(updateData: snapshot, currentTimeUnix: timestampSeconds)
+  if res.isOk() {
+    print("RGS initialized for the first time")
+  }
+}
+
+// Get current snapshot from the RGS Server
+func getSnapshot(lastSyncTimeStamp: UInt32) -> [UInt8]? {
+  // Use LDK's RGS Server or use your own Server
+  let url: URL = URL(string: "https://rapidsync.lightningdevkit.org/snapshot/\(lastSyncTimeStamp)")!
+  let data = // Use the url to get the data
+  if let data = data {
+    return [UInt8](data)
+  }
+  return nil
+}
+```
+
+</template>
+
+</CodeSwitcher>
+
+**Implementation notes:** this struct is not required if you are providing your own routes. It will be used internally in `ChannelManager` to build a `NetworkGraph`. Other networking options are: `LDKNetwork_Bitcoin`, `LDKNetwork_Regtest` and `LDKNetwork_Testnet`
+
+**Dependencies:** `Logger`
+
+**Optional dependency:** `Access`, a source of chain information. Recommended to be able to verify channels before adding them to the internal network graph.
+
+**References:** [Rust `P2PGossipSync` docs](https://docs.rs/lightning/*/lightning/routing/gossip/struct.P2PGossipSync.html), [`Access` docs](https://docs.rs/lightning/*/lightning/chain/trait.Access.html), [Java/Kotlin `P2PGossipSync` bindings](https://github.com/lightningdevkit/ldk-garbagecollected/blob/main/src/main/java/org/ldk/structs/P2PGossipSync.java), [Rust `RapidGossipSync` docs](https://docs.rs/lightning-rapid-gossip-sync/*/lightning_rapid_gossip_sync/), [Java/Kotlin `RapidGossipSync` bindings](https://github.com/lightningdevkit/ldk-garbagecollected/blob/main/src/main/java/org/ldk/structs/RapidGossipSync.java)
+
+### Optional: Initialize `Probabilistic Scorer`
+
+**What it's used for**: to find a suitable payment path to reach the destination.
+
+<CodeSwitcher :languages="{rust:'Rust', kotlin:'Kotlin', swift:'Swift'}">
+
+<template v-slot:rust>
+
+```rust
+let network_graph_path = format!("{}/network_graph", ldk_data_dir.clone());
+let network_graph = Arc::new(disk::read_network(Path::new(&network_graph_path), args.network, logger.clone()));
+
+let scorer_path = format!("{}/scorer", ldk_data_dir.clone());
+let scorer = Arc::new(RwLock::new(disk::read_scorer(
+    Path::new(&scorer_path),
+    Arc::clone(&network_graph),
+    Arc::clone(&logger),
+)));
+
+```
```

Comment:
This is using utilities from the sample, which aren't available in LDK. Similarly above for gossip sync.

Reply:
For the sake of keeping the code snippets short maybe it's ok?